File: DataEngineering/05_Stream_Processing/projects/golang/04_intermediate_kafka_client/producer.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/segmentio/kafka-go"
)

func main() {
	topic := "events-topic"
	partition := 0

	// Dial the leader of the partition
	conn, err := kafka.DialLeader(context.Background(), "tcp", "localhost:9092", topic, partition)
	if err != nil {
		log.Fatalf("failed to dial leader: %v", err)
	}
	defer conn.Close()

	// Produce 10 messages
	for i := 1; i <= 10; i++ {
		msg := fmt.Sprintf("Event %d", i)
		_, err := conn.WriteMessages(
			kafka.Message{Value: []byte(msg)},
		)
		if err != nil {
			log.Fatalf("failed to write messages: %v", err)
		}
		fmt.Printf("Produced message: %s\n", msg)
		time.Sleep(500 * time.Millisecond) // Simulate delay between events
	}
}

File: Tool-Kit/Kafka/examples/01_basic_producer_consumer/producer.py
from confluent_kafka import Producer
import time

def delivery_report(err, msg):
    if err is not None:
        print(f"Message delivery failed: {err}")
    else:
        print(f"Message delivered to {msg.topic()} [{msg.partition()}] at offset {msg.offset()}")

conf = {'bootstrap.servers': 'localhost:9092'}
producer = Producer(conf)

for i in range(10):
    data = f"Message {i}"
    producer.produce('basic-topic', key=str(i), value=data, callback=delivery_report)
    producer.poll(0)
    time.sleep(0.5)

producer.flush()
